### PR TITLE
test: match_ingredient_origin unit test

### PR DESCRIPTION
**Pull request by alexgarel**
Testing match_ingredient_origin as a way to understand bug cited on pro platform

part of:
- #8173

### CODE REVIEW

File: tests/unit/expected_test_results/parse_origins_from_text/simple-extraction-fr.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+   "lc" : "fr",
+   "origin_fr" : "Sucre France."
+}
```
Comment (alexgarel) [inline tests/unit/expected_test_results/parse_origins_from_text/simple-extraction-fr.json:4]: This is a bug, isn't it @stephanegigandet ?
We should have specific_ingredients

File: tests/unit/expected_test_results/parse_origins_from_text/real-list-fr.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+   "lc" : "fr",
+   "origin_fr" : "amandes d’Espagne. Chocolat noir de Côte d’Ivoire. Huile de noisette d’Italie. sucre de France. noisettes d’Italie. fèves de cacao de Côte d’Ivoire. Framboises lyophilisées d’Espagne. arôme naturel de framboise fabriqué en France.",
+   "specific_ingredients" : [
+      {
+         "id" : "en:sugar",
+         "ingredient" : "sucre",
+         "origins" : "en:france",
+         "text" : "sucre de France."
+      }
```
Comment (alexgarel) [inline tests/unit/expected_test_results/parse_origins_from_text/real-list-fr.json:10]: This is a bug, isn't it @stephanegigandet ?
We should have a lot more origins in specific_ingredients

File: tests/unit/match_ingredient_origin.t [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			"amandes d’Espagne. Chocolat noir de Côte d’Ivoire. Huile de noisette d’Italie. sucre de France. noisettes d’Italie. fèves de cacao de Côte d’Ivoire. Framboises lyophilisées d’Espagne. arôme naturel de framboise fabriqué en France.",
+		expected => {
+			"amandes" => "Espagne",
+			"Chocolat noir" => "Côte d’Ivoire",
+			"Huile de noisette" => "Italie",
+			"sucre" => "France",
+			"noisettes" => "Italie",
+			"fèves de cacao" => "Côte d’Ivoire",
+			"Framboises lyophilisées" => "Espagne" . "arôme naturel de framboise" => "France",
+		},
```
Comment (alexgarel) [inline tests/unit/match_ingredient_origin.t:61]: I'm not sure it's exactly this, but something like it. (right now it fails, of course)

**Comment by stephanegigandet** [comment]
@alex There were a few issues. One was the lacking support for different single quotes (’ vs ')

I added support for "Ingredient Country" (e.g. "Sugar France"), but only if Sugar is a known ingredient.

Is this ok for you?

**Comment by alexgarel** [comment]
Very good @stephanegigandet !

But as I am the one who opened the PR you must approve it :-)

**Review by stephanegigandet — APPROVED** [review]

**Pull request merged**